### ISSUE TITLE
[TopAppBar] Icon buttons in Card have the class `mdc-icon-button` while the buttons in Top App Bar don't

### ISSUE THREAD

**Issue by tychenjiajun** [material-components/material-components-web#4527]
It's strange and confusing.
[Card](https://material.io/develop/web/components/cards/)
```html
<div class="mdc-card__action-icons">
   <button class="material-icons mdc-icon-button mdc-card__action mdc-card__action--icon" title="Share">share</button>
    <button class="material-icons mdc-icon-button mdc-card__action mdc-card__action--icon" title="More options">more_vert</button>
  </div>
```

[Top App Bar](https://material.io/develop/web/components/top-app-bar/)
```html
<section class="mdc-top-app-bar__section mdc-top-app-bar__section--align-end" role="toolbar">
      <a href="#" class="material-icons mdc-top-app-bar__action-item" aria-label="Download" alt="Download">file_download</a>
      <a href="#" class="material-icons mdc-top-app-bar__action-item" aria-label="Print this page" alt="Print this page">print</a>
      <a href="#" class="material-icons mdc-top-app-bar__action-item" aria-label="Bookmark this page" alt="Bookmark this page">bookmark</a>
    </section>
```

**Comment by moog16** [comment]
@tychenjiajun the Card is just a wrapper component, which means we didn't build in icon styles. This means we have to use mdc-icon-button in the Card. 

However looking at `@mixin mdc-top-app-bar-icon_` is very similar to the `@mixin mdc-icon-button-base_`. And this might be a good opprotunity to combine the 2 mixins and use mdc-icon-button instead and remove the `mdc-top-app-bar-icon` mixin. I will wait for others on the team to chime in.

**Comment by moog16** [comment]
We will look into making top app bar support icon button, and possibly deprecate the `mdc-top-app-bar-icon_` mixin in the future.

**Comment by moog16** [comment]
Closing. Merged #4745
